website/content/docs/concepts/events.mdx

```diff
@@ -0,0 +1,150 @@
+| kv     | `kv-v2/metadata-delete` | 1.13          |
+| kv     | `kv-v2/metadata-patch`  | 1.13          |
+| kv     | `kv-v2/metadata-read`   | 1.13          |
+| kv     | `kv-v2/metadata-write`  | 1.13          |
+| kv     | `kv-v2/undelete`        | 1.13          |
+
+
+## Event Format
+
+Events may be formatted in protobuf binary format or as JSON.
+See `EventReceived` in [`sdk/logical/event.proto`](https://github.com/hashicorp/vault/blob/main/sdk/logical/event.proto) in the relevant Vault version for the protobuf schema.
+
+- `event` `(Event)` - contains the event that happened.
+
+  - `id` `(string)` - unique identifier for the event. The `id` is unique for each event, and events with the same `id` represent the same event.
+
+  - `metadata` `(object)` - arbitrary extra data customized for the event type.
+
+- `eventType` `(string)` - the event type that was published.
+
+- `pluginInfo` `(PluginInfo)` - information about the plugin that generated the event, if applicable.
+
+  - `mountClass` `(string)` - the class of plugin, e.g., `secret`, `auth`.
+
+  - `mountAccessor` `(string)` - the unique ID of the mounted plugin.
+
+  - `mountPath` `(string)` - the path that the plugin is mounted at.
+
+  - `plugin` `(string)` - the name of the plugin, e.g., `kv`.
+
+- `timestamp` `(string)` - ISO 8601-formatted timestamp for when the event was generated.
+
+Here is an example event in JSON format:
+
+```json
+{
+    "event": {
+        "id": "827e2cc0-6ace-42f1-2517-409616ce1fbc",
+        "metadata": {
+            "current_version": "2",
+            "oldest_version": "0",
+            "path": "data/secure123"
+        }
+    },
+    "eventType": "kv-v2/data-write",
+    "pluginInfo": {
+        "mountClass": "secret",
+        "mountAccessor": "kv_6dd6f6c6",
+        "mountPath": "secret/",
+        "plugin": "kv"
+    },
+    "timestamp": "2023-02-16T21:57:34.719199Z"
+}
+```
+
+## Subscribing to Events
+
+Vault has an API endpoint, `/v1/sys/events/subscribe/{eventType}`, that allows users to subscribe to events via a
+WebSockets stream.
+This endpoint supports the standard authenticaion and authorization workflows used by other Vault endpoints.
+The `{eventType}` parameter is a non-empty string of what event type to subscribe, which may contain wildcards (`*`)
+to subscribe to multiple events, e.g., `kv-v2/data-*`.
+
+By default, the events delivered in protobuf binary format.
+The endpoint can also format the data as JSON if the `json` query parameter is set to `true`:
+
+```shell-session
+$ wscat -H "X-Vault-Token: $(vault print token)" --connect 'ws://127.0.0.1:8200/v1/sys/events/subscribe/kv-v2/data-write?json=true
+{"event":{"id":"827e2cc0-6ace-42f1-2517-409616ce1fbc", "metadata":{"current_version":"2", "oldest_version":"0", "path":"data/secure123"}}, "eventType":"kv-v2/data-write", "pluginInfo":{"mountClass":"secret", "mountAccessor":"kv_6dd6f6c6", "mountPath":"secret/", "plugin":"kv"}, "timestamp":"2023-02-16T21:57:34.719199Z"}
```

Comment:
As I'm reading this, I'm realising we use `underscore_separated` rather than `camelCase` for serialising JSON fields elsewhere in the Vault API. Are we able to make that consistent automatically using the protobuf definitions?

Reply:
I've opened https://github.com/hashicorp/vault/pull/19246 for this.